File: src/src/org/github/olloginov/ideataskbrowser/tasks/FetchNewIssuesTask.java
package org.github.olloginov.ideataskbrowser.tasks;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.progress.ProgressIndicator;
import com.intellij.openapi.progress.ProgressManager;
import com.intellij.openapi.progress.Task;
import com.intellij.openapi.project.Project;
import org.github.olloginov.ideataskbrowser.TaskBrowserBundle;
import org.github.olloginov.ideataskbrowser.view.TaskSearchTreeNode;
import org.github.olloginov.ideataskbrowser.view.TaskTreeModel;
import org.github.olloginov.ideataskbrowser.view.TreeNodeRef;
import org.jetbrains.annotations.NotNull;

public class FetchNewIssuesTask extends Task.Backgroundable {
    private final TaskTreeModel treeModel;

    public FetchNewIssuesTask(@NotNull Project project, @NotNull TaskTreeModel treeModel) {
        super(project, TaskBrowserBundle.message("FetchNewIssuesTask.title"), true);
        this.treeModel = treeModel;
    }

    @Override
    public void run(@NotNull ProgressIndicator indicator) {
        TaskTreeModel.RootTreeNode rootTreeNode = treeModel.getRoot();
        for (int index = 0; index < rootTreeNode.getChildCount(); ++index) {
            final TreeNodeRef<TaskSearchTreeNode> target = new TreeNodeRef<>(treeModel, rootTreeNode.getChildAt(index));
            ApplicationManager.getApplication().invokeLater(new Runnable() {
                @Override
                public void run() {
                    ProgressManager.getInstance().run(new FetchNewIssuesFromRepoTask(myProject, target));
                }
            });
        }
    }
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/tasks/FetchNewIssuesTask.kt
package org.github.olloginov.ideataskbrowser.tasks

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.progress.ProgressIndicator
import com.intellij.openapi.progress.ProgressManager
import com.intellij.openapi.project.Project
import org.github.olloginov.ideataskbrowser.TaskBrowserBundle
import org.github.olloginov.ideataskbrowser.view.TaskTreeModel
import org.github.olloginov.ideataskbrowser.view.TreeNodeRef

class FetchNewIssuesTask(
    project: Project,
    private val treeModel: TaskTreeModel
) : com.intellij.openapi.progress.Task.Backgroundable(project, TaskBrowserBundle.message("fetchNewIssuesTask.title"), true) {

    override fun run(indicator: ProgressIndicator) {
        val rootTreeNode = treeModel.root
        for (index in 0 until rootTreeNode.childCount) {
            val target = TreeNodeRef(treeModel, rootTreeNode.getChildAt(index))
            ApplicationManager.getApplication().invokeLater {
                ProgressManager.getInstance().run(FetchNewIssuesFromRepoTask(project, target))
            }
        }
    }
}
